### ISSUE TITLE
Difficulty in making compile environment.

### ISSUE THREAD

**Issue by kjleeic** [wichtounet/dll#4]
I'm sorry to have to ask too many questions.

I'm a hardware engineer who researches about A.I. chip.

I really want to use your DLL framework for my research, but I have some trouble in making compile environment.

My environments is:

> OS: Ubuntu 14.04
> g++: 4.9
> clang++: 3.4 **(Installing clang++3.7 in Ubuntu 14.04 is hard..I try to various way but it doesn't work)**
> CUDA: cuda-7.5
> cuDNN: v5 for cuda-7.5
> 2 GPU 

My Makefile is:

> DLL_HOME        = /home/kjlee/lib/dll
> CUDA_HOME       = /usr/local/cuda-8.0
> 
> CLANG           = clang++
> CXXFLAGS        += -std=c++14
> LDFLAGS         += -lpthread
> 
> 
> ##########################################################
> ### DLL
> ##########################################################
> CPPFLAGS        += -I${DLL_HOME}/include
> CPPFLAGS        += -I${DLL_HOME}/Catch/include
> CPPFLAGS        += -I${DLL_HOME}/etl/include
> CPPFLAGS        += -I${DLL_HOME}/etl/lib/include
> CPPFLAGS        += -I${DLL_HOME}/mnist/include
> CPPFLAGS        += -I${DLL_HOME}/cifar-10/include
> CPPFLAGS        += -I${DLL_HOME}/nice-svm/include
> 
> DLLFLAGS        += -DETL_PARALLEL
> DLLFLAGS        += -DETL_VECTORIZE_FULL
> ##DLLFLAGS      += -DETL_BLAS_MODE
> DLLFLAGS        += -DETL_CUBLAS_MODE
> DLLFLAGS        += -DETL_CUDNN_MODE
> ##DLLFLAGS      += -DETL_EGBLAS_MODE
> 
> 
> ##########################################################
> ### CUDA
> ##########################################################
> CPPFLAGS        += -I${CUDA_HOME}/include
> LDFLAGS         += -L${CUDA_HOME}/lib64 -lcublas -lcudart
> 
> 
> 
> ##########################################################
> ### MAKE
> ##########################################################
> all:
> 
> main:
>         ${CLANG} ${CXXFLAGS} ${CPPFLAGS} ${LDFLAGS} ${DLLFLAGS} mnist_ae.cpp
> 
> 

I want to know your environment.
If you use Ubuntu 14.04, please let me know how to install clang++3.7.

Thanks.

**Comment by wichtounet** [comment]
Hi, 

Thanks for your interest. 

Don't be sorry. If the library is too hard to use, it's my fault. 

There was a mistake in the Readme, the versions were not up to date. This has been fixed in ee9e6a0

Personally, I'm doing my tests on Gentoo with G++ 6.3 and 7.1 and Clang 3.9 and clang 4.0. All these four compilers are known to work.

If installing clang is not convenient, you should install a more recent version of GCC. Installing GCC 6 is not complicated on Ubuntu: 

```
sudo apt-get update
sudo apt-get install build-essential software-properties-common
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt-get update
sudo apt-get install gcc-snapshot
sudo apt-get update
sudo apt-get install gcc-6 g++-6
```

After this, you will have a g++-6 executable installed on your machine and it should work for DLL. 

If you really prefer clang, you can take a look at this (follow 14.04 or trusty example): https://askubuntu.com/questions/787383/how-to-install-llvm-3-9

I hope that helps.

**Comment by wichtounet** [comment]
Seems you resolved your issue, I'll close this issue. Let a comment if you still have a problem.
